src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorServerReadyPublisher.cs

```diff
@@ -15,11 +15,13 @@ internal class RazorServerReadyPublisher : ProjectSnapshotChangeTrigger
         private readonly ForegroundDispatcher _foregroundDispatcher;
         private ProjectSnapshotManagerBase _projectManager;
         private readonly IClientLanguageServer _languageServer;
+        private readonly ClientNotifierService _clientNotifierService;
         private bool _hasNotified = false;
 
         public RazorServerReadyPublisher(
             ForegroundDispatcher foregroundDispatcher,
-            IClientLanguageServer languageServer)
+            IClientLanguageServer languageServer,
```

Comment:
Do you still need the IClientLanguageServer anymore?

Reply:
I needed to DI the IClientLanguageServer but not the tcs, I was just hacking something together so it didn't occur to me to use the DI factory.